src/services/study/network-modifications.js

```diff
@@ -1084,6 +1084,57 @@ export function createSubstation(
     });
 }
 
+/**
+ * Formats the properties of an array of properties so it can be consumed by the backend.
+ * @param {Array<{key: string, value: any}>} previousPropertiesArray - The previous propeties va.
+ * @param {Array<{key: string, value: any}>} newPropertiesArray - The new properties values.
+ * @returns {Array<{name: string, value: any, previousValue: any, added: boolean, deletionMark: boolean}>} - The modified properties.
+ */
+export function formatPropertiesForBackend(
+    previousPropertiesArray,
+    newPropertiesArray
+) {
+    const propertiesModifications = [];
+
+    previousPropertiesArray.forEach((oldObj) => {
+        const newObj = newPropertiesArray.find(
+            (newObj) => newObj.name === oldObj.name
+        );
+
+        if (!newObj) {
+            propertiesModifications.push({
+                ...oldObj,
+                previousValue: oldObj.value,
+                value: null,
+                // added: false,
+                deletionMark: true,
+            });
+        } else if (newObj.value !== oldObj.value) {
+            propertiesModifications.push({
+                ...newObj,
+                added: false,
+                deletionMark: false,
+                previousValue: oldObj.value,
+            });
+        }
+    });
+
+    newPropertiesArray.forEach((newObj) => {
+        const oldObj = previousPropertiesArray.find(
+            (oldObj) => oldObj.name === newObj.name
+        );
+
+        if (!oldObj) {
+            propertiesModifications.push({
+                ...newObj,
+                added: true,
+                deletionMark: false,
+            });
+        }
+    });
```

Comment:
Maybe you can refactor this by creating  a single method and use it also  for previousPropertiesArray.

Reply:
i didn't understand what part do you mean? this one
```
propertiesModifications.push({
                ...newObj,
                added: true,
                deletionMark: false,
            });
```

or only this part ? 
```
const updatedProperty = newPropertiesArray.find(
            (updatedObj) => updatedObj.name === previousPropertiePair.name
        );
```